### PR TITLE
Merge create and update model to get read only properties

### PR DESCRIPTION
**Pull request by anshikg**
*Issue #, if available:*

*Description of changes:* Merging create model with update model to make sure we are fetching the read only properties.

*Test:*
```
(env) anshikg@3c22fbb25c23 aws-cloudwatch-alarm % cfn test --enforce-timeout 120                             
[Warning] Resource spec validation would fail from next major version. Provider should mark additionalProperties as false if the property is of object type and has properties or patternProperties defined in it. Please fix the warnings: 'additionalProperties' is a required property

Failed validating 'required' in schema['properties']['properties']['patternProperties']['^[A-Za-z0-9]{1,64}$']['allOf'][0]['dependencies']['patternProperties']:
    {'$comment': 'An object cannot have both defined and undefined '
                 'properties; therefore, properties is not allowed when '
                 'patternProperties is specified. Provider should mark '
                 'additionalProperties as false if the property is of '
                 'object type and has patternProperties defined in it.',
     'not': {'required': ['properties']},
     'required': ['additionalProperties']}

On instance['properties']['ExtendedStatistic']:
    {'description': 'The percentile statistic for the metric associated '
                    'with the alarm. Specify a value between p0.0 and '
                    'p100.',
     'patternProperties': {'^p(\\d{1,2}(\\.\\d{0,2})?|100)$': {'type': 'string'}},
     'type': 'string'}
Resource schema is valid.
=================================================================================================================== test session starts ====================================================================================================================
platform darwin -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /Users/anshikg/workspace/cloudformation-cli/env/bin/python3
cachedir: .pytest_cache
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm/.hypothesis/examples')
rootdir: /Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm, configfile: ../../../../../private/var/folders/zh/rhw_046j7bq4d88hjn671hqjy323j8/T/pytest_nl27pxob.ini
plugins: random-order-1.0.4, localserver-0.5.0, cov-2.11.1, hypothesis-6.8.1
collected 16 items                                                                                                                                                                                                                                         

handler_create.py::contract_create_delete PASSED                                                                                                                                                                                                     [  6%]
handler_create.py::contract_invalid_create PASSED                                                                                                                                                                                                    [ 12%]
handler_create.py::contract_create_duplicate PASSED                                                                                                                                                                                                  [ 18%]
handler_create.py::contract_create_read_success PASSED                                                                                                                                                                                               [ 25%]
handler_create.py::contract_create_list_success PASSED                                                                                                                                                                                               [ 31%]
handler_delete.py::contract_delete_read PASSED                                                                                                                                                                                                       [ 37%]
handler_delete.py::contract_delete_list PASSED                                                                                                                                                                                                       [ 43%]
handler_delete.py::contract_delete_update PASSED                                                                                                                                                                                                     [ 50%]
handler_delete.py::contract_delete_delete PASSED                                                                                                                                                                                                     [ 56%]
handler_delete.py::contract_delete_create PASSED                                                                                                                                                                                                     [ 62%]
handler_misc.py::contract_check_asserts_work PASSED                                                                                                                                                                                                  [ 68%]
handler_read.py::contract_read_without_create PASSED                                                                                                                                                                                                 [ 75%]
handler_update.py::contract_update_read_success PASSED                                                                                                                                                                                               [ 81%]
handler_update.py::contract_update_list_success PASSED                                                                                                                                                                                               [ 87%]
handler_update_invalid.py::contract_update_create_only_property PASSED                                                                                                                                                                               [ 93%]
handler_update_invalid.py::contract_update_non_existent_resource PASSED         
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by prerna-p — CHANGES_REQUESTED** [review]
Do we need all readOnly properties in the request? We've also had discussions in the past around this:
https://github.com/aws-cloudformation/cloudformation-cli/issues/626#issuecomment-726373464

**Comment by anshikg** [comment]
> Do we need all readOnly properties in the request? We've also had discussions in the past around this:
> [#626 (comment)](https://github.com/aws-cloudformation/cloudformation-cli/issues/626#issuecomment-726373464)

Thats good point. Let me check if we can prune just the read only properties vs all the properties

File: src/rpdk/core/contract/resource_client.py [hunk 1]
```diff
@@ -69,6 +69,24 @@ def prune_properties_if_not_exist_in_path(output_model, input_model, paths):
     return output_document["properties"]
 
 
+def prune_properties_which_dont_exist_in_path(model, paths):
+    """Prune given properties from a model. This method removes any property
+    from the model which does not exists in the paths
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function returns the model after pruning the path which exists
+    in the paths tuple but not in the input_model
```
Comment (prerna-p) [inline src/rpdk/core/contract/resource_client.py:78]: method description needs improvement. I think it is pruning properties which exist in path and returning those.

File: tests/contract/test_resource_client.py [hunk 2]
```diff
@@ -1210,10 +1226,12 @@ def test_generate_invalid_update_example_with_inputs(resource_client_inputs):
 
 def test_generate_update_example_with_primary_identifier(resource_client_inputs_schema):
     created_resource = resource_client_inputs_schema.generate_create_example()
```
Comment (prerna-p) [inline tests/contract/test_resource_client.py:1228]: recommending to test locally by running contract test against a resource.

Comment (anshikg) [reply tests/contract/test_resource_client.py:1228]: updated the contract test updated in the details section

File: src/rpdk/core/contract/resource_client.py (hunk 1 above)
Comment (anshikg) [reply src/rpdk/core/contract/resource_client.py]: update the description

**Review by prerna-p — APPROVED** [review]

File: src/rpdk/core/contract/resource_client.py [hunk 3]
```diff
@@ -69,6 +69,24 @@ def prune_properties_if_not_exist_in_path(output_model, input_model, paths):
     return output_document["properties"]
 
 
+def prune_properties_which_dont_exist_in_path(model, paths):
+    """Prune given properties from a model. This method removes any property
```
Comment (priyap286) [inline src/rpdk/core/contract/resource_client.py:73]: "Prunes model to properties present in path." seems easier to understand to me than "Prune given properties from a model."

File: src/rpdk/core/contract/resource_client.py [hunk 4]
```diff
@@ -426,7 +444,12 @@ def generate_update_example(self, create_model):
             if self.create_only_paths:
                 self.validate_update_example_keys(unique_identifiers, update_example)
             update_example.update(unique_identifiers)
-            return update_example
+            create_model_with_read_only_properties = (
+                prune_properties_which_dont_exist_in_path(
```
Comment (priyap286) [inline src/rpdk/core/contract/resource_client.py:448]: We are successfully including top level read only properties what about sub-properties which are populated with default values post creation?

Comment (priyap286) [reply src/rpdk/core/contract/resource_client.py:448]: Spoke offline. Will be addressed in another PR.

**Review by priyap286 — APPROVED** [review]

**Pull request merged**